app/utils/navigators/PartnershipNavigator.scala

```diff
@@ -116,4 +124,29 @@ class PartnershipNavigator @Inject()(val dataCacheConnector: UserAnswersCacheCon
       case _ => NavigateTo.dontSave(controllers.routes.SessionExpiredController.onPageLoad())
     }
   }
-}
\ No newline at end of file
+
+  private def regionBasedNameNavigation(answers: UserAnswers): Option[NavigateTo] = {
+    answers.get(AreYouInUKId) match {
+      case Some(false) => NavigateTo.dontSave(routes.PartnershipRegisteredAddressController.onPageLoad())
+      case _ => NavigateTo.dontSave(routes.ConfirmPartnershipDetailsController.onPageLoad())
+    }
+  }
+
+  private def regionBasedNavigation(answers: UserAnswers): Option[NavigateTo] = {
+    answers.get(PartnershipRegisteredAddressId) flatMap { address =>
+      countryOptions.regions(address.country.getOrElse("")) match {
+        case UK => NavigateTo.dontSave(controllers.register.routes.BusinessTypeAreYouInUKController.onPageLoad(CheckMode))
+        case EuEea => NavigateTo.dontSave(routes.WhatYouWillNeedController.onPageLoad())
+        case RestOfTheWorld => NavigateTo.dontSave(routes.OutsideEuEeaController.onPageLoad())
+        case _ => NavigateTo.dontSave(controllers.routes.SessionExpiredController.onPageLoad())
+      }
+    }
+  }
+
+  private def regionBasedContactDetailsRoutes(answers: UserAnswers): Option[NavigateTo] = {
+    answers.get(AreYouInUKId) match {
+      case Some(false) => NavigateTo.save(routes.CheckYourAnswersController.onPageLoad())
+      case _ => NavigateTo.save(routes.PartnershipVatController.onPageLoad(NormalMode))
```

Comment:
case _ could be Some(true) or None - should the behaviour be the same for both cases?

Reply:
fixed